### PR TITLE
Unify npm run build and create_dist commands.

### PR DESCRIPTION
**Pull request by goodov**
The change is to unify both commands to have a single `build` command internally, which is called as `build -- --target=create_dist` when doing `npm run create_dist`.
 
Test release builds successful https://ci.brave.com/job/test-brave-browser-build/126/

Resolves https://github.com/brave/brave-browser/issues/40497

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: build/commands/scripts/commands.js [hunk 1]
```diff
@@ -135,6 +132,28 @@ program
...
+  .option('--no_gn_gen', 'Build without running gn gen')
+  .option('--notarize', 'notarize targets that support it with Apple')
+  .option('--offline', 'use offline mode for RBE')
+  .option('--prepare_only', 'Do not build targets, but prepare everything (build redirect_cc, update branding, etc.)')
+  .option('--skip_signing', 'skip signing binaries')
+  .option('--tag_ap <ap>', 'ap for stub/standalone installer')
+  .option('--tag_installdataindex <index>', 'installdataindex for stub/standalone installer')
+  .option('--target <target>', 'Comma-separated list of targets to build, instead of the default browser target')
+  .option('--target_android_base <target_android_base>', 'target Android SDK level for apk or aab (classic, modern, mono)', 'classic')
+  .option('--target_android_output_format <target_android_output_format>', 'target Android output format (apk, aab)', 'aab')
```
Comment (goodov) [inline build/commands/scripts/commands.js:152]: this is the only thing that will be different after this PR is merged. It's Android-specific, so cc @SergeyZhukovsky 

Before this unification, this flag was empty by default in the `build` command but defaulted to `aab` in the `create_dist` command. Now, it defaults to `aab` in both the `build` and `create_dist` commands.

It's set in CI everywhere, so we might consider leaving it empty for local development if that's more convenient.

**Review by fmarier — APPROVED** [review]
`package.json` change :+1:

**Review by kylehickinson — APPROVED** [review]
iOS didn't use `create_dist` at all, and regular build command looks the same so lgtm!

**Review by atuchin-m — APPROVED** [review]

File: build/commands/scripts/commands.js (hunk 1 above)
Comment (goodov) [reply build/commands/scripts/commands.js:152]: cc @samartnik @SergeyZhukovsky ^

should we have `--target_android_output_format` default to `aab` in `build/create_dist` command or should we default it to empty string, which will internally fallback to `apk`?

Comment (samartnik) [reply build/commands/scripts/commands.js:152]: Yes, `aab` in both `build` and `create_dist` commands is ok. `aab` is primary format now, `apk` is for devs, which they can set manually for local builds.

**Review by bridiver — APPROVED** [review]

**Pull request merged**